test/extensions/filters/http/fault/fault_filter_test.cc

```diff
@@ -312,6 +312,58 @@ TEST_F(FaultFilterTest, AbortWithHttpStatus) {
   EXPECT_EQ("fault_filter_abort", decoder_filter_callbacks_.details_);
 }
 
+TEST_F(FaultFilterTest, AbortWithHttpStatusRuntimeOverride) {
```

Comment:
I don't think we need all these new tests, I would just add a direct test on the settings object for each override.